contracts/upgradeable-polymesh-ink/src/lib.rs

```diff
@@ -324,81 +345,95 @@ upgradable_api! {
                     )
                     .submit()?;
 
-                // Create settlement.
                 api.call()
                     .settlement()
                     .execute_manual_instruction(instruction_id, None, fungible, nfts, offchain, None)
                     .submit()?;
                 Ok(())
             }
 
-            /// Asset issue tokens.
+            /// Issues `amount_to_issue` new tokens to `portfolio_kind`.
             #[ink(message)]
-            pub fn asset_issue(&self, ticker: Ticker, amount: Balance, portfolio: PortfolioKind) -> PolymeshResult<()> {
+            pub fn asset_issue(
+                &self,
+                asset_id: AssetID,
+                amount_to_issue: Balance,
+                portfolio_kind: PortfolioKind
+            ) -> PolymeshResult<()> {
                 let api = Api::new();
-                // Mint tokens.
-                api.call().asset().issue(ticker, amount, portfolio).submit()?;
+                api.call().asset().issue(asset_id, amount_to_issue, portfolio_kind).submit()?;
                 Ok(())
             }
 
-            /// Asset redeem tokens.
+            /// Redeems `amount_to_redeem` tokens from `portfolio_kind`.
             #[ink(message)]
-            pub fn asset_redeem(&self, ticker: Ticker, amount: Balance, portfolio: PortfolioKind) -> PolymeshResult<()> {
+            pub fn asset_redeem(
+                &self,
+                asset_id: AssetID,
+                amount_to_redeem: Balance,
+                portfolio_kind: PortfolioKind
+            ) -> PolymeshResult<()> {
                 let api = Api::new();
-                // Redeem tokens.
-                api.call().asset().redeem_from_portfolio(ticker, amount, portfolio).submit()?;
+                api.call().asset().redeem(asset_id, amount_to_redeem, portfolio_kind).submit()?;
                 Ok(())
             }
 
-            /// Very simple create asset and issue.
+            /// Creates a new asset and issues `amount_to_issue` tokens of that asset to the default portfolio of the caller.
             #[ink(message)]
-            pub fn asset_create_and_issue(&self, name: AssetName, ticker: Ticker, asset_type: AssetType, divisible: bool, issue: Option<Balance>) -> PolymeshResult<()> {
+            pub fn asset_create_and_issue(
+                &self,
+                asset_name: AssetName,
+                asset_type: AssetType,
+                divisible: bool,
+                amount_to_issue: Option<Balance>
+            ) -> PolymeshResult<()> {
                 let api = Api::new();
-                // Create asset.
+
+                let contract_account = ink::env::account_id::<PolymeshEnvironment>();
+                let nonce = api.query().asset().asset_nonce(contract_account)?;
+                let asset_id =
+                    blake2_128(&(b"modlpy/pallet_asset", contract_account, nonce).encode());
```

Comment:
We should provide a helper method that contracts can call, since they might need the `asset_id` to make other calls.